core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala

```diff
@@ -82,9 +82,8 @@ class CheckpointFile[T](val file: File, version: Int, formatter: CheckpointFileF
         line = reader.readLine()
         if (line == null)
           return Seq.empty
-        val version = line.toInt
-        version match {
-          case version =>
+        line.toInt match {
+          case `version` =>
```

Comment:
The previous version was correct because it matched against CurrentVersion. The backticks are only necessary if the variable name starts with lowercase. A bit magical, I know.